api/python/provisioner/srv/salt/repos/upgrade.sls

```diff
@@ -15,14 +15,23 @@
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 #
 
-{% macro repo_mounted(release, source, mount_dir) %}
+{% from './_macros.sls' import setup_repos with context %}
 
-sw_upgrade_repo_iso_mounted_{{ release }}:
-  mount.mounted:
-    - name: {{ mount_dir }}
-    - device: {{ source }}
-    - mkmnt: True
-    - fstype: iso9660
-    - persist: True
+{% set base_dir = pillar['release']['upgrade']['base_dir'] %}
 
-{% endmacro %}
+{% for version, repos in pillar['release']['upgrade']['repos'].items() %}
+
+    {% set single_iso_repo = {version: repos.pop(version)} %}
+
+    # to mount the single ISO first
+    {{ setup_repos(single_iso_repo, base_dir) }}
+
+    {% set upgrade_repos = dict() %}
+    {% for release in repos %}
+    {% set key =  'sw_upgrade_' ~ release ~ '_' ~ version %}
```

Comment:
minor thing: indentation of loop's inner-logic will make that easier to understand